pareto/examples/multiobjective_optimization/seismicity_vs_cost_MOO.ipynb

```diff
@@ -0,0 +1,531 @@
+    "fixed_risk_vals.append(min_cost_risk)\n",
+    "cost_vals_fixed_risk.append(min_cost_obj)\n",
+    "\n",
+    "min_cost_for_zero_risk = cost_vals_fixed_risk[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "326a34c6-1982-44c0-8b02-22ab1cfa7645",
+   "metadata": {},
+   "source": [
+    "We can plot the results as a Pareto curve (note that the concept of Pareto optimality inspired the name for Project PARETO):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "78ae7fa7-38ab-41e1-b9d5-0dd340b46229",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Create plot of Pareto curve\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "fig, ax = plt.subplots()\n",
+    "scatter = ax.scatter(cost_vals_fixed_risk, fixed_risk_vals, c=\"blue\")\n",
+    "ax.set_xlabel(f\"Cost [{model.v_Z.get_units()}]\")\n",
+    "ax.set_ylabel(f\"Subsurface risk [{model.v_Z_SubsurfaceRisk.get_units()}]\")\n",
+    "\n",
+    "print(f\"Cost values [{model.v_Z.get_units()}]: {cost_vals_fixed_risk}\")\n",
+    "print(\n",
+    "    f\"Subsurface risk values [{model.v_Z_SubsurfaceRisk.get_units()}]: {fixed_risk_vals}\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "83943b23-08f3-467c-8041-6100afd6878d",
+   "metadata": {},
+   "source": [
+    "The Pareto curve shows that compared to the pure minimum cost solution, we can significantly reduce the subsurface risk with nearly negligible increases in the cost.\n",
+    "\n",
+    "The epsilon constraint method can also be used by minimizing risk for fixed values of cost:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b7b0ca67-8b3b-4d37-8a20-3ef56f6462cb",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Change objective function back to minimum subsurface risk\n",
+    "set_objective(model, Objectives.subsurface_risk)\n",
+    "\n",
+    "# Unfix subsurface risk variable\n",
+    "model.v_Z_SubsurfaceRisk.unfix()\n",
+    "\n",
+    "# Create lists for results\n",
+    "fixed_cost_vals = list(range(6200, 7800, 100))  # Cost risk values to iterate over\n",
+    "risk_vals_fixed_cost = list()  # List to store corresponding subsurface risk values\n",
+    "\n",
+    "for c in fixed_cost_vals:\n",
+    "    print(\"*\" * 60)\n",
+    "    print(f\"Solving minimum subsurface risk constraining cost to {c}\")\n",
+    "    print(\"*\" * 60)\n",
+    "    model.v_Z.fix(c)  # Fix the cost to the specified value\n",
+    "    results_min_risk = solve_model(model=model, options=options)\n",
+    "\n",
+    "    assert results_min_risk.solver.termination_condition == TerminationCondition.optimal\n",
+    "    assert results_min_risk.solver.status == SolverStatus.ok\n",
+    "\n",
+    "    with nostdout():\n",
+    "        feasibility_status = is_feasible(model)\n",
+    "\n",
+    "    assert feasibility_status\n",
+    "\n",
+    "    risk_vals_fixed_cost.append(model.v_Z_SubsurfaceRisk.value)\n",
```

Comment:
`model.v_Z_SubsurfaceRisk.value - model.subsurface.y_dist.value`

Reply:
I changed the subsurface risk objective function so that this suggestion is not necessary.